code/game/mecha/working/ripley.dm

```diff
@@ -72,6 +73,7 @@
 	slow_pressure_step_in = 3 //step_in while in normal pressure conditions
 	step_in = 3
 	resistance_flags = LAVA_PROOF | FIRE_PROOF | ACID_PROOF
+	weather_protection = WEATHER_LAVA|WEATHER_STORM
```

Comment:
are you actually checking mech's weather_protection for weather_lava on lava itself?

Reply:
It's for the floor is lava weather type which isn't actually lava

Reply:
Ah, right